src/main/java/com/hillel/items_exchange/controller/CategoryController.java

```diff
@@ -21,86 +24,63 @@
 
 @RestController
 @RequestMapping("/category")
-@CrossOrigin(origins = "http://localhost:4200")
 @RequiredArgsConstructor
-@Slf4j
 public class CategoryController {
 
     private final CategoryService categoryService;
 
     @GetMapping("/names")
-    public ResponseEntity<List<String>> getAllCategoriesNames() {
-        List<String> categoriesNames = categoryService.findAllCategoryNames();
-        if (categoriesNames.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
-
-        return new ResponseEntity<>(categoriesNames, HttpStatus.OK);
+    @ApiOperation(value = "Get all names of existing categories.")
+    public ResponseEntity<List<String>> getAllCategoryNames() {
+        return new ResponseEntity<>(categoryService.findAllCategoryNames(), HttpStatus.OK);
     }
 
     @GetMapping("/all")
+    @ApiOperation(value = "Get all existing categories.")
     public ResponseEntity<List<CategoryDto>> getAllCategories() {
-        List<CategoryDto> categories = categoryService.findAllCategories();
-        if (categories.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
-
-        return new ResponseEntity<>(categories, HttpStatus.OK);
+        return new ResponseEntity<>(categoryService.findAllCategoryDto(), HttpStatus.OK);
     }
 
     @GetMapping("/{category_id}")
+    @ApiOperation(value = "Get a category by its ID, if it exists")
     public ResponseEntity<CategoryDto> getCategoryById(@PathVariable("category_id") long id) {
-        return categoryService.findCategoryById(id)
+        return categoryService.findCategoryDtoById(id)
                 .map(categoryDto -> new ResponseEntity<>(categoryDto, HttpStatus.OK))
                 .orElseThrow(() -> new EntityNotFoundException(getExceptionMessageSourceWithId(
                         id, "invalid.category.id")));
     }
 
     @PreAuthorize(HAS_ROLE_ADMIN)
     @PostMapping
-    public ResponseEntity<CategoryDto> createCategory(@Valid @RequestBody CategoryDto categoryDto) {
-        if (isCategoryIdValidForCreating(categoryDto)
-                && categoryService.isCategoryNameHasNotDuplicate(categoryDto.getName())) {
-
-            return new ResponseEntity<>(categoryService.addNewCategory(categoryDto), HttpStatus.CREATED);
+    @ApiOperation(value = "Save the new category if its and internal subcategory IDs are zero. Only for users authorized as admin.")
+    public ResponseEntity<CategoryDto> createCategory(@Validated(New.class) @RequestBody CategoryDto categoryDto) {
+        if (categoryService.isCategoryDtoValidForCreating(categoryDto)) {
+            return new ResponseEntity<>(categoryService.create(categoryDto), HttpStatus.CREATED);
```

Comment:
The same approach (ResponseStatus) could be used here as well.

Reply:
Got it.